### ISSUE TITLE
[CS 1.6] Map type specific items doesn't appear on VGUI Menu when map prefix doesn't coincide

### ISSUE THREAD

**Issue by GiovaniFerraroTrivelli** [ValveSoftware/halflife#2614]
When you use the VGUI Menu, if you play a map with defusal objective, if it doesn't have de_ in its name, you can't buy Defusal Kit using the VGUI Menu. You need to enable the old menu to buy the item or buy it from console.

Same thing happens when you play a VIP objective map. You can't select VIP from the jointeam (3) because this item doesn't appear (while in old menu it does).

How to test:
- rename de_dust2 to dust2 or simply change its prefix to something different to de_.
- play as_oilrig and then play [vip_truck](https://maps.cs-bg.info/maps/cs/3022/), after choosing CT as team.

Tested in beta build and old build and this happens in both versions.
```
Protocol version 48
Exe version 1.1.2.7/Stdio (cstrike)
Exe build: 10:18:33 Jul 19 2019 (8303)
```
```
Protocol version 47
Exe version 1.1.2.5 (cstrike)
Exe build: 10:56:59 Oct  3 2005 (326
```

**Comment by basuritashka** [comment]
Good catch, this is indeed happening when renaming maps with `de_*` prefix, `Defusal Kit` wont be available to buy through VGUI Menu as it will appear as not available.

![image](https://user-images.githubusercontent.com/33007491/61600826-b5b49900-ac00-11e9-9309-0a2c9de4dc47.png)

**Comment by SamVanheer** [comment]
The `IsDEMapType` and `IsASMapType`  functions on the client are checking this.

It might be a good idea to make the server inform the client whether functionality for these maps should be enabled, but since some maps may be relying on this behavior now there should be a way to disable it to make it behave like it does now.

**Comment by CS-PRO1** [comment]
Same happens with most of CS:GO ports maps (css_nuke/csg_dust2 etc..). I believe the game should check if the map has a bombsite entity to enable defuse kits or some other way (that's all I can think of now lmao)

**Comment by SamVanheer** [comment]
> 
> 
> Same happens with most of CS:GO ports maps (css_nuke/csg_dust2 etc..). I believe the game should check if the map has a bombsite entity to enable defuse kits or some other way (that's all I can think of now lmao)

The server already checks that, so if it can tell the client it should be good. The problem is that may require new network messages, and old clients won't have those. Depending on how those old clients handle non-existent messages it may or may not be a problem.

**Comment by CS-PRO1** [comment]
> The problem is that may require new network messages, and old clients won't have those. Depending on how those old clients handle non-existent messages it may or may not be a problem.

Those damn non-steam clients.. for how long they're going to block the netcode from being fixed up? Enough is enough...

**Comment by wilianmaique** [comment]
> > The problem is that may require new network messages, and old clients won't have those. Depending on how those old clients handle non-existent messages it may or may not be a problem.
> 
> Those damn non-steam clients.. for how long they're going to block the netcode from being fixed up? Enough is enough...

The largest audience is cs no-steam. while on steam it has 10-15k no-steam has 30-50k.

I am in favor of leaving cs 1.6 free as well as csgo.

**Comment by Maxi605** [comment]
> > > The problem is that may require new network messages, and old clients won't have those. Depending on how those old clients handle non-existent messages it may or may not be a problem.
> > 
> > 
> > Those damn non-steam clients.. for how long they're going to block the netcode from being fixed up? Enough is enough...
> 
> The largest audience is cs no-steam. while on steam it has 10-15k no-steam has 30-50k.
> 
> I am in favor of leaving cs 1.6 free as well as csgo.

Can't, not like this, not now, needs a ton of work and improvement on the anticheat system.

besides Steam doesn't support Windows XP and CS is popular as it needs lower resources

**Comment by CS-PRO1** [comment]
They can literally re-crack the newer releases easily. They must follow us not that we have to care about them anyway..

**Comment by tute123456** [comment]
Hello, I have a problem, when I want to play with the bots, only one map that I play on a server appears to choose from in the list of maps, all the others do not appear, I only have 2 options: "Ramdom Map" and the map from the server, I already looked at the "maps" folder and they are all there, even when I enter with ramdon map and change the map, (changelevel of_aztec for example) it loads everything without problems, any ideas?
